The terraform-provider-azurerm_v2.71.0_x5.exe plugin crashed!

azurerm_iothub_dps.iot-dps: Creating...
╷
│ Error: Plugin did not respond
│
│   with azurerm_iothub_dps.iot-dps,
│   on dps.tf line 1, in resource "azurerm_iothub_dps" "iot-dps":
│    1: resource "azurerm_iothub_dps" "iot-dps" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call.  
│ The plugin logs may contain more details.
╵

 

Stack trace from the terraform-provider-azurerm_v2.71.0_x5.exe plugin:

 

panic: interface conversion: interface {} is string, not iothub.AllocationPolicy

 

goroutine 132 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub.resourceIotHubDPSCreateUpdate(0xc0005e6c80, 0x5f19de0, 0xc0005c2e00, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/iothub/iothub_dps_resource.go:181 +0x11ec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000c6b880, 0x6ec28e8, 0xc001906100, 0xc0005e6c80, 0x5f19de0, 0xc0005c2e00, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x1f6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000c6b880, 0x6ec28e8, 0xc001906100, 0xc000da3570, 0xc0005a7300, 0x5f19de0, 0xc0005c2e00, 0x0, 0x0, 0x0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000005d58, 0x6ec28e8, 0xc001906100, 0xc0017ccf50, 0xc001906100, 0x6401a00, 0xc000d0d600)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc001cbffe0, 0x6ec2990, 0xc001906100, 0xc000da3340, 0xc001cbffe0, 0xc000d0d680, 0xc0016d9ba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb6
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x6401a00, 0xc001cbffe0, 0x6ec2990, 0xc000d0d680, 0xc0009f2c60, 0x0, 0x6ec2990, 0xc000d0d680, 0xc001908500, 0x483)      
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x222        
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00027c8c0, 0x6f0f8d8, 0xc000187800, 0xc00169de60, 0xc00097eab0, 0xa4c8ee0, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00027c8c0, 0x6f0f8d8, 0xc000187800, 0xc00169de60, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000756200, 0xc00027c8c0, 0x6f0f8d8, 0xc000187800, 0xc00169de60)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

 

Error: The terraform-provider-azurerm_v2.71.0_x5.exe plugin crashed!

 

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/internal/services/iothub/iothub_dps_resource.go
+++ b/internal/services/iothub/iothub_dps_resource.go
@@ -178,7 +178,7 @@ func resourceIotHubDPSCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) 
 		Sku:      expandIoTHubDPSSku(d),
 		Properties: &iothub.IotDpsPropertiesDescription{
 			IotHubs:          expandIoTHubDPSIoTHubs(d.Get("linked_hub").([]interface{})),
-			AllocationPolicy: d.Get("allocation_policy").(iothub.AllocationPolicy),
+			AllocationPolicy: iothub.AllocationPolicy(d.Get("allocation_policy").(string)),
 		},
 		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
@@ -250,7 +250,7 @@ func resourceIotHubDPSRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		d.Set("service_operations_host_name", props.ServiceOperationsHostName)
 		d.Set("device_provisioning_host_name", props.DeviceProvisioningHostName)
 		d.Set("id_scope", props.IDScope)
-		d.Set("allocation_policy", props.AllocationPolicy)
+		d.Set("allocation_policy", string(props.AllocationPolicy))
 	}
 
 	return tags.FlattenAndSet(d, resp.Tags)

